pkg/network/types/types.go

```diff
@@ -22,9 +22,10 @@ const (
 // that the node will publish publicly
 // this is used to be able to configure public side of a node
 type IfaceInfo struct {
-	Name    string   `json:"name"`
-	Addrs   []IPNet  `json:"addrs"`
-	Gateway []net.IP `json:"gateway"`
+	Name       string           `json:"name"`
+	Addrs      []IPNet          `json:"addrs"`
+	Gateway    []net.IP         `json:"gateway"`
+	MacAddress net.HardwareAddr `json:"macaddress"`
```

Comment:
net.HardwareAddr does not have marshal methods afaik, so are we sure this is going to properly print in json? (`xx:xx:xx:xx:xx:xx`)?